.github/actions/generate_sitemap_and_robots_new/action.yml

```diff
@@ -0,0 +1,35 @@
+name: "Generate Sitemap.xml and Robots.txt"
+description: "Generate Sitemap.xml and robots.txt for all domains and upload to R2"
+inputs:
+  main_domain:
+    description: "Main domain for sitemap replacement"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Process and combine sitemaps
+      shell: bash
+      env:
+        MAIN_DOMAIN: ${{ inputs.main_domain }}
+      run: |
+        mkdir -p content
+        domain_name=$(echo $MAIN_DOMAIN | sed 's|https://||g')
+        mkdir -p content/$domain_name
+
+        # Process and combine sitemaps
+        node .github/modify_sitemap_new.js \
+          --staging-sitemap temp/staging_sitemap.xml \
+          --academy-sitemap temp/academy_sitemap.xml \
+          --output-file content/$domain_name/sitemap_new.xml \
+          --new-domain $domain_name
+
+        # Copy robots.txt and update it
+        cp temp/robots.txt content/$domain_name/robots.txt
```

Comment:
this part I don't understand. so inside `.github/workflows/generate_sitemap_new.yml` we download the robots.txt from `https://staging.deriv.com/robots.txt` and copy it to `temp/robots.txt` and here we again copy it to `content/$domain_name/robots.txt` and then we never use it. 
I think what needs to be done is to add a new argument to `.github/modify_robots_new.js` and then pass `content/$domain_name/robots.txt` to it. Then, instead of hardcoding `robotsContent`, the script should read the content from this file, apply any necessary modifications, and write the result to the output file.